Partial correlations

fredmanglis opened the pull request:
#### Description
Implements functions necessary for partial correlations

#### How should this be tested?
Running the unit tests. The migrated functions have had unit tests added with them.

#### Any background context you want to provide?
https://github.com/genenetwork/gn-gemtext-threads/blob/main/topics/gn1-migration-to-gn2/partial-correlations.gmi

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)
N/A

#### Questions
N/A

fredmanglis committed:
Migrate `export_informative` function

Issue:
https://github.com/genenetwork/gn-gemtext-threads/blob/main/topics/gn1-migration-to-gn2/partial-correlations.gmi

* gn3/partial_correlations.py: Implement a mostly, bug-compatible
  `export_informative` function as part of migrating the partial correlations
  feature over to GN3 from GN1
* tests/unit/test_partial_correlations.py: Implement tests to ensure the code
  work in a similar manner as that one in GN1.

fredmanglis committed:
Move 'export_trait_data' to 'gn3.db.traits' module

Issue:
https://github.com/genenetwork/gn-gemtext-threads/blob/main/topics/gn1-migration-to-gn2/partial-correlations.gmi

* gn3/db/traits.py: Move function `export_trait_data` here
* gn3/heatmaps.py: Remove function `export_trait_data`
* tests/unit/db/test_traits.py: Move function `export_trait_data` tests here
* tests/unit/test_heatmaps.py: Remove function `export_trait_data` here

  Function `export_trait_data` more closely corresponds to the traits and is
  used in more than just the `gn3.heatmaps` module. This commit moves the
  relevant code over to the `gn3.db.traits` module and also moves the tests to
  the corresponding tests modules.

fredmanglis committed:
Move `export_informative` function to `gn3.db.traits` module

Issue:
https://github.com/genenetwork/gn-gemtext-threads/blob/main/topics/gn1-migration-to-gn2/partial-correlations.gmi

* gn3/db/traits.py: Move `export_informative` function here
* gn3/partial_correlations.py: Remove `export_informative` function
* tests/unit/db/test_traits.py: Move `export_informative` function tests here
* tests/unit/test_partial_correlations.py: Remove `export_informative`
  function tests

  The `export_informative` function relates more to the traits than to the
  partial correlations, and could find use in more than just the partial
  correlations stuff. This commit moves the function to the more
  traits-specific `gn3.db.traits` module.

fredmanglis committed:
Implement `control_samples` function as is in GN1

Issue:
https://github.com/genenetwork/gn-gemtext-threads/blob/main/topics/gn1-migration-to-gn2/partial-correlations.gmi

* gn3/partial_correlations.py: Implement `control_samples` function
* tests/unit/test_partial_correlations.py: add tests for `control_samples`
  function

  Implement the function `control_samples` and make it mostly bug-compatible
  with the `web/webqtl/correlation/correlationFunction.controlStrain` function
  in GN1.

  This implementation in GN3 does not do any calls to the database. It will
  rely on other functions to provide the data from the database to it.

fredmanglis commented:
@BonfaceKilz @arunisaac While there is more work to still be done regarding the partial correlations, I have decided to start the pull/merge request early, so as to avoid having a huge amount of code for review. Smaller chunks of code review should be better, methinks, and will get the code into the system piecemeal.

WDYT?

fredmanglis committed:
Disable pylint issue

* Disable minor pylint issue.

fredmanglis committed:
Implement `dictify_by_samples`

Issue:
https://github.com/genenetwork/gn-gemtext-threads/blob/main/topics/gn1-migration-to-gn2/partial-correlations.gmi

* gn3/partial_correlations.py: implement `dictify_by_samples` function
* tests/unit/test_partial_correlations.py: implement tests for
  `dictify_by_samples` function

  Implement the `dictify_by_samples` function as a partial migration of the
  `web.webqtl.correlation.correlationFunction.fixStrains` function from GN1.

BonfaceKilz reviewed (commented):
This LGTM. Should I go ahead and merge?

fredmanglis committed:
Implement remaining `fix_samples` functionality

Issue:
https://github.com/genenetwork/gn-gemtext-threads/blob/main/topics/gn1-migration-to-gn2/partial-correlations.gmi

* gn3/partial_correlations.py: implement `fix_samples` function
* tests/unit/test_partial_correlations.py: implement tests for `fix_samples`
  function

  Implement the remaining partial migration for the
  `web.webqtl.correlation.correlationFunction.fixStrain` function in GN1.

fredmanglis committed:
Merge branch 'main' of github.com:genenetwork/genenetwork3 into partial-correlations

fredmanglis commented:
Yes please, but don't delete the branch yet... There's still  more work to
be done on this.

On Tue, 19 Oct 2021, 10:12 BonfaceKilz, ***@***.***> wrote:

> ***@***.**** commented on this pull request.
>
> This LGTM. Should I go ahead and merge?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork3/pull/45#pullrequestreview-782871229>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMIL2MUIJJHDPDVHEN3OU3UHUK5HANCNFSM5GGMXL3Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

BonfaceKilz commented:
Muriithi Frederick Muriuki
***@***.***> anaandika:

> Yes please, but don't delete the branch yet... There's still  more work to
> be done on this.
>
Cool. You have a local branch though. Once you are
done though, remember to delete the branch ;)

> On Tue, 19 Oct 2021, 10:12 BonfaceKilz, ***@***.***> wrote:
>
>> ***@***.**** commented on this pull request.
>>
>> This LGTM. Should I go ahead and merge?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/genenetwork/genenetwork3/pull/45#pullrequestreview-782871229>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAMIL2MUIJJHDPDVHEN3OU3UHUK5HANCNFSM5GGMXL3Q>
>> .
>> Triage notifications on the go with GitHub Mobile for iOS
>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android
>> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>>
>>

-- 
Bonface M. K.
D4F09EB110177E03C28E2FE1F5BBAE1E0392253F (hkp://keys.gnupg.net)
Free Software Activist
Humble GNU Emacs User | Bearer of scheme-y parens
Curator: <https://upbookclub.com> | Twitter: @BonfaceKilz
